Begin.py/begin.py

```diff
@@ -52,76 +49,85 @@ def __repr__(self):
         s += 'MIJN EIGENSCHAP:\n' + str(self.my_feature)
         return s
 
-
     # Voeg hier andere methodes toe.
     # Je hebt in het bijzonder methodes nodig die het model vullen.
 
     # METHODS
-    def read_text_from_file(self, filename):
+
+    @staticmethod
+    def read_text_from_file(filename):
         """
-        methode:    read_text_from_file(self, filename) 
+        methode:    read_text_from_file(self, filename)
         argument:   self; verwijst naar de klasse TextModel en maakt hiervan 'a instance of a class'
-        argument:   filename; van het type string. 
-                    self.text: variabele waaraan de inhoud toegekend is 
-                    van het bestand filename. Type: 1 hele lange string.   
+        argument:   filename; van het type string.
+                    self.text: variabele waaraan de inhoud toegekend is
+                    van het bestand filename. Type: 1 hele lange string.
         """
 
-        with open("/home/annemarleen/programming/CS_for_all/Eindopdracht/test.txt") as f:
-            self.text = f.read().replace("\n", "").rstrip("")            
+        file = open(filename)
+        text = file.read().replace("\n", "").rstrip("")
+
+        return text
 
-        return self.text
-        
-    
-  
     def make_sentence_lengths(self, text):
-        """        
+        """
         De make_sentence_lengths(self) moet parameter text (type string) gebruiken om de
         dictionary self.sentence_lengths te vullen.
         output: dictionary self.sentence_lengths.
-        Het resultaat van deze functie is: dictionary: sentence; words 
+        Het resultaat van deze functie is: dictionary: sentence; words
 
         assert tm.sentence_lengths == {16: 1, 5: 1, 6: 1} => 1 zin met 16 woorden, 1 zin met 5 woorden, 1 zin met 6 woorden
-        
+
         """
         # dus, je moet het aantal zinnen tellen, én het aantal woorden in de zin.
-        # als x is zin, return True. If True, count words. if word endswhith '.?!' dan 
+        # als x is zin, return True. If True, count words. if word endswhith '.?!' dan
         # count_words. If word endswith '.?!' dan return is_sentence. count_sentence =+ 1
         # het resultaat hiervan wordt dmv een dictionary getoond.
 
-        print("The whole original string is:\n",self.text)
+        print("The whole original string is:\n", text)
         print("")
-        
-        pw = "$"
-        LoW = self.text.split()
-        count = 0
-        zin = []
-
-        #print(LoW)         
-
-        for nw in LoW:
-            if nw not in ".?!" or pw == "$":
-                count +=1
-            if nw[-1] in ".?!":
-              #  pw = "$" 
-                zin += [count]
-                count = 0
+
+        previous_word = "$"
```

Comment:
altijd gewoon leesbare variabelen gebruiken. als je ooit op de schaal gaat werken dat de grootte van je python code uit gaat maken dan heb je allang geleerd om de code met een compiler te draaien.
je geschreven code is voor mensen om aan te werken en te kunnen lezen, je kan een compiler gebruiken om systeemcode te maken te bedoeld is voor machines.